frontEnd/tickets-app/src/pages/InfoFilm/InfoFilm.js

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from "react";
+import { Grid, Stack } from "@mui/material";
+import { filmsApi } from "../../api/filmsApi";
+import { useParams } from "react-router-dom";
+import "./InfoFilms.scss";
+import DatePicker, { registerLocale } from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import ru from "date-fns/locale/ru";
+import MovieSession from "../../components/MovieSession/MovieSession";
+import { usefullFunctions } from "../../usefullFunction/usefullFunctions";
+import { parseISO } from "date-fns";
+import { io } from "socket.io-client";
+import ShowSeats from "../../components/ShowSeats/ShowSeats";
+
+const InfoFilm = () => {
+  registerLocale("ru", ru);
+
+  const [film, setFilm] = useState([]);
+  const [date, setDate] = useState(
+    usefullFunctions.onlyDayMonthYear(Date.now(), true)
+  );
+  const [filmInfo, setFilmInfo] = useState({});
+  const [showSession, setShowSession] = useState({});
+  const [isShowSession, setIsShowSession] = useState(false);
+  const [seats, setSeats] = useState(false);
+  const [socket, setSocket] = useState(false);
+  const param = useParams();
+
+  const getArrFilms = async (newDate) => {
+    let response = await filmsApi.getMovie(param.id, newDate);
+    const sessions = {};
+
+    response.data.forEach((e) => {
+      if (e.cinema.name in sessions) {
```

Comment:
you should put this complicated logic to helper functions and put them into a separated file